Lib/pdb.py

```diff
@@ -2077,34 +2075,36 @@ def help():
 
 
 def main():
-    import getopt
+    import argparse
 
-    opts, args = getopt.getopt(sys.argv[1:], 'mhc:', ['help', 'command='])
+    parser = argparse.ArgumentParser(prog="pdb",
+                                     description=_usage,
+                                     formatter_class=argparse.RawDescriptionHelpFormatter)
 
-    if not args:
-        print(_usage)
-        sys.exit(2)
+    parser.add_argument('-c', '--command', action='append', default=[])
+    parser.add_argument('-m', action='store_true')
+    parser.add_argument('pyfile', nargs=1)
+    parser.add_argument('args', nargs="*")
 
-    if any(opt in ['-h', '--help'] for opt, optarg in opts):
-        print(_usage)
-        sys.exit()
+    if len(sys.argv) == 1:
```

Comment:
Not sure if this check is necessary since parse_args() would check the argument validity and print the usage/exit if invalid.

Reply:
This is to mimic the old behavior as close as possible - when the user do `python -m pdb`, it'll print the full help message(including the description like `-h`), instead of reporting an error of missing required arguments.

Reply:
argparse provides better error messages, no? Why not taking its enhancements?

Reply:
> argparse provides better error messages, no? Why not taking its enhancements?

```
python -m pdb
```

Current:
```
usage: pdb [-h] [-c command] (-m module | pyfile) [args ...]

Debug the Python program given by pyfile. Alternatively,
an executable module or package to debug can be specified using
the -m switch.

Initial commands are read from .pdbrc files in your home directory
and in the current directory, if they exist.  Commands supplied with
-c are executed after commands from .pdbrc files.

To let the script run until an exception occurs, use "-c continue".
To let the script run up to a given line X in the debugged file, use
"-c 'until X'".

positional arguments:
  pyfile
  args

options:
  -h, --help            show this help message and exit
  -c command, --command command
  -m module
```

Without the if check:
```
usage: pdb [-h] [-c command] (-m module | pyfile) [args ...]
pdb: error: one of the arguments -m pyfile is required
```

Reply:
Ah you're right, it looks better. Maybe add a comment to explain why you do that.